File: jegl/src/ch/std/jegl/image/CharImage.java
/*
 * see https://www.simtech-ag.ch/education/java/jegl/resources/basic/for
 * author: Daniel Schmutz
 * copyright: Simtech AG (https://www.simtech-ag.ch)
 */
package ch.std.jegl.image;

public class CharImage {

	public static void main(String[] args) {
		final int xDim = 10;
		final int yDim = 10;
		for(int i = 0; i < xDim; i++) {
			for(int j = 0; j < yDim; j++) {
				if (i == j) {
					System.out.print('x');				
				} else {
					System.out.print("&");								
				}
			}
			System.out.println();
		}
	}
}

File: jjas/basic/for/CharImage.js
/*
 * see https://www.simtech-ag.ch/education/internet/jjas/resources/basic/for
 * author: Daniel Schmutz
 * copyright: Simtech AG (https://www.simtech-ag.ch)
*/

console.log('Start CharImage ...');

const xDim = 10;
const yDim = 10;
for(let i = 0; i < xDim; i++) {
    let line = '';
	for(let j = 0; j < yDim; j++) {
        if (i == j) {
            line += 'x';	
        } else {
            line += '&';								
        }
    }
    console.log(line);
}
console.log('... CharImage Done');
